### PR TITLE
Fix broken anchor links

### PR DESCRIPTION
**Pull request by colinrotherham**
Found a few broken anchor links whilst clicking around:

1. [**/community/**#community-principles](https://design-system.service.gov.uk/community/#community-principles)
2. [**/get-started/extending-and-modifying-components/**#avoid-overwriting-gov-uk-design-system-code](https://design-system.service.gov.uk/get-started/extending-and-modifying-components/#avoid-overwriting-gov-uk-design-system-code)
3. [**/styles/spacing/**#spacing-override-classes](https://design-system.service.gov.uk/styles/spacing/#spacing-override-classes)
4. [**/styles/spacing/**#spacing-on-custom-components](https://design-system.service.gov.uk/styles/spacing/#spacing-on-custom-components)

Realised our link checker wasn't checking anchors so switched to [hyperlink](https://www.npmjs.com/package/hyperlink) (used by [govuk-frontend-docs](https://github.com/alphagov/govuk-frontend-docs))

### CODE REVIEW

File: src/components/error-summary/default/index.njk [hunk 1]
```diff
@@ -10,11 +10,11 @@ layout: layout-example.njk
   errorList: [
     {
       text: "The date your passport was issued must be in the past",
-      href: "#passport-issued-day"
+      href: "#"
```
Comment (colinrotherham) [inline src/components/error-summary/default/index.njk:13]: These were handy to show "real" examples but only `#` skips missing anchors

**Review by CharlotteDowns — APPROVED** [review]
I can't speak for the other files but the proposed redirect for these broken links seems right. Thank you for picking up.

**Comment by querkmachine** [comment]
I suspect the use of `#1`, `#2`, `#3`, etc. in the header and footer examples was so that clicking on one didn't mark every link as having been visited, but I'm not aware of anything that cares about that distinction (we don't have `:visited` styles on those links, for example), so I've no objection to changing them.

**Review by querkmachine — COMMENTED** [review]
LGTM. 

I'm a little on the fence about the removal of the anchors in the error summary and skiplink examples—I feel it makes them look a lot more like generic placeholders, rather than being explicit that these should always be in-page anchors—but I figure we can always add them back if we find that it's creating misunderstandings.

**Review by querkmachine — APPROVED** [review]

**Pull request merged**